File: AbstracFactory/Python/Fabricas.py
from abc import abstractclassmethod
import Piezas

class AbstrasFactory:
    @abstractclassmethod
    def CreateScreen():
        pass
    @abstractclassmethod
    def CreateCamera():
        pass
    @abstractclassmethod
    def CreateCase():
        pass

class PhoneH(AbstrasFactory):
    @staticmethod
    def CreateScreen():
        return Piezas.ScreenOled()
    @staticmethod
    def CreateCamera():
        return Piezas.CameraUp()
    @staticmethod
    def CreateCase():
        return Piezas.CarcasaAcero()

class PhoneL(AbstrasFactory):
    @staticmethod
    def CreateScreen():
        return Piezas.ScreenLCD()
    @staticmethod
    def CreateCamera():
        return Piezas.CameraDown()
    @staticmethod
    def CreateCase():
        return Piezas.CarcasaPlastico()

File: AbstracFactory/CPP/Fabricas.cpp
#include <bits/stdc++.h>
#include "Piezas.cpp"

using namespace std;

class Fabricas{
    private:

    public:
        virtual Screen* CreateScreen()=0;
        virtual Camera* CreateCamera()=0;
        virtual Case* CreateCase()=0;
};

// Clase de telefono 1 
class PhoneH:public Fabricas{
    private:

    public:
        virtual Screen* CreateScreen();
        virtual Camera* CreateCamera();
        virtual Case* CreateCase();
};

Screen* PhoneH::CreateScreen(){
    Screen* s=new ScreenOled();
    return s;
}
Camera* PhoneH::CreateCamera(){
    Camera* c=new CameraUp();
    return c;
}
Case* PhoneH::CreateCase(){
    Case* c=new CaseAcero();
    return c;
}

// Clase de telefono 2 
class PhoneL:public Fabricas{
    private:

    public:
        virtual Screen* CreateScreen();
        virtual Camera* CreateCamera();
        virtual Case* CreateCase();
};

Screen* PhoneL::CreateScreen(){
    Screen* s=new ScreenLCD();
    return s;
}
Camera* PhoneL::CreateCamera(){
    Camera* c=new CameraDown();
    return c;
}
Case* PhoneL::CreateCase(){
    Case* c=new CasePlastico();
    return c;
}